ui/src/views/Scopes/ViewScope/index.jsx

```diff
@@ -63,20 +59,46 @@ export default class ViewScope extends Component {
   };
 
   handleSearchSubmit = searchTerm => {
-    this.setState({ searchTerm });
+    const { location, history } = this.props;
+    const query = parse(location.search.slice(1));
+
+    if (query.searchTerm !== searchTerm) {
+      const newQuery = {
+        ...query,
+        searchTerm,
+      };
+
+      history.push({
+        search: stringify(newQuery, { addQueryPrefix: true }),
+      });
+    }
   };
 
   handleTabChange = (event, value) => {
-    this.setState({ currentTabIndex: value });
+    const { location, history } = this.props;
+    const query = parse(location.search.slice(1));
+
+    if (query.currentTabIndex !== value) {
+      const newQuery = {
+        ...query,
+        currentTabValue: value,
+      };
+
+      history.push({
+        search: stringify(newQuery, { addQueryPrefix: true }),
+      });
+    }
   };
 
   render() {
     const {
       classes,
+      location,
       match: { params },
       data: { loading, error, clients, roles },
     } = this.props;
-    const { searchTerm, currentTabIndex } = this.state;
+    const query = parse(location.search.slice(1));
+    const { searchTerm = '', currentTabIndex = 0 } = query;
```

Comment:
Noticed a possible regression:

STR:

1. Navigate to http://localhost:5080/auth/scopes/test?searchTerm=reality&currentTabValue=1

Expected: The tab focus is on Clients
Actual: The tab focus is on Roles

<img width="1765" alt="Screen Shot 2020-04-22 at 4 08 24 PM" src="https://user-images.githubusercontent.com/3766511/80028691-81ce1300-84b3-11ea-8c56-e42c15d6155c.png">